### PR TITLE
Switch to kotlinx.ast.

### LINKED ISSUE #24: Migrate to kotlinx.ast from kastree
Kastree is currently unsupported and occasionally results in a `java.lang.NoClassDefFoundError: com/sun/jna/Native` error. 

We should move to https://github.com/kotlinx/ast as that is the recommended library and has active development. 

This can be worked around by adding `implementation("net.java.dev.jna:jna:+")` to your dependencies.

### PR DESCRIPTION
**Pull request by yanks**
Fixes #24 + facilitates a Java 11 Move.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -7,6 +7,7 @@ plugins {
 }
 
 repositories {
+    maven { url 'https://jitpack.io' }
```
Comment (digitalbuddha) [inline build.gradle:10]: what needed jitpack? In general I'm not big on jitpack for upstream dependencies. AFAIK jitpack builds from source and goes down if github is down, similar it does not have the same security guarantees something like maven central or google repo has. Can we shade (include the jar in lib) whatever the dependecy is? I can also try to find a different one that fulfills same use case OR I can repackage under my own domain and throw on maven central.

File: src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt [hunk 2]
```diff
@@ -264,31 +263,62 @@ class RealHyperShard(
...
+        val parseResult = KotlinGrammarAntlrKotlinParser.parseKotlinFile(source)
+        val allDeclarations = parseResult.summary(attachRawAst = false).get()
+
+        // Find header
+        val classIdentifier = fun (): String {
+            for (declaration in allDeclarations) {
+                val packageHeader = declaration as? PackageHeader ?: continue
+                return packageHeader.identifier.joinToString(separator = ".") { it.identifier }
+            }
+            throw Exception("Expected a PackageHeader attribute in source.")
```
Comment (digitalbuddha) [inline src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt:276]: nit unspecific exception type, maybe go with ParseException?

File: src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt [hunk 3]
```diff
@@ -264,31 +263,62 @@ class RealHyperShard(
...
-            val shouldProcessTestMethods = shouldProcessTestMethods(ktClass, annotationValue,
-                notAnnotationValue)
             if (!shouldProcessTestMethods) {
                 continue
             }
 
-            for (element in declaration.children) {
-                val ktClassBody = element as? KtClassBody ?: continue
+            for (element in ktClass.children) {
+                if ((element as? DefaultAstNode)?.description != "classBody") {
```
Comment (digitalbuddha) [inline src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt:297]: nit, this reads a bit dense, sanity check if element is not default as node do you want this to evaluate to true?

File: src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt [hunk 4]
```diff
@@ -324,15 +357,21 @@ class RealHyperShard(
...
+                    if (ktNamedFunction.keyword != "fun") {
+                        continue
+                    }
                     // the fqName includes the test name with the full package name and test name
                     // all delimited by periods, this allows the test to be ran because we need
                     // the hashtag
-                    for (fnAnnotationEntry in ktNamedFunction.annotationEntries) {
-                        if (fnAnnotationEntry.shortName.toString() == "Test") {
+                    for (fnAnnotationEntry in ktNamedFunction.annotations) {
+                        if (fnAnnotationEntry.identifier.first().identifier == "Test") {
```
Comment (digitalbuddha) [inline src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt:374]: should this be firstOrNull(), could identifier be an empty collection?

File: build.gradle (hunk 1 above)
Comment (yanks) [reply build.gradle:10]: Yeah; I think what I'm going to end up doing instead is just including a built-jar inside this repo. kotlinx.ast isn't published anywhere else.

File: src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt (hunk 4 above)
Comment (yanks) [reply src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt:374]: Probably! Thanks

**Pull request merged**